src/domain/core/src/services/export_service.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::{Display, Formatter};
+use std::path::Path;
+
+use datafusion::dataframe::DataFrame;
+use internal_error::{BoxedError, InternalError};
+use thiserror::Error;
+
+use crate::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Service
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait ExportService: Send + Sync {
+    async fn export_to_fs(
+        &self,
+        df: DataFrame,
+        path: &Path,
```

Comment:
What confuses a bit is that is not simpliy a path, rather a prefix of the path.
Maybe some comments?

Reply:
it's a path. either path to file or to directory. if it's a file path (something with extension and without trailing path delimiter), DataFusion will store data into it. Otherwise, it treats path as a dir and will create files (with generated names, and extension related to export format) inside.